src/pybind/mgr/orchestrator/module.py

```diff
@@ -1875,7 +1875,7 @@ def _apply_nvmeof(self,
             raise OrchestratorValidationError('unrecognized command -i; -h or --help for usage')
 
         spec = NvmeofServiceSpec(
-            service_id=f'{pool}.{group}',
+            service_id=f'{pool}.{group}' if group else pool,
```

Comment:
It could be a good idea to match legacy service naming when empty string '' group is being used.

```python
Python 3.12.4 (main, Jun  6 2024, 18:26:44) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>>
>>> group=''
>>> pool='rbd'
>>> service_id = f"{pool}.{group}" if group else pool
>>> service_id
'rbd'
```